analyzer/src/main/kotlin/managers/utils/PubCacheReader.kt

```diff
@@ -64,14 +63,26 @@ internal class PubCacheReader {
             .find { !it.isSymbolicLink() && it.isFile && it.name == filename }
     }
 
-    fun findProjectRoot(packageInfo: JsonNode): File? {
+    fun findProjectRoot(packageInfo: JsonNode, workingDir: File): File? {
         val packageVersion = packageInfo["version"].textValueOrEmpty()
         val type = packageInfo["source"].textValueOrEmpty()
         val description = packageInfo["description"]
         val packageName = description["name"].textValueOrEmpty()
         val url = description["url"].textValueOrEmpty()
         val resolvedRef = description["resolved-ref"].textValueOrEmpty()
+        val resolvedPath = description["path"].textValueOrEmpty()
+        val isPathRelative = description["relative"].booleanValue()
+
+        if (type == "path" && resolvedPath.isNotEmpty()) {
```

Comment:
I have created a separate `if` with a separate return because the return statement at line 104 makes operations on `path` that are not compatible with the operations I need to perform in case `path` is a local directory.